### ISSUE TITLE
Batch convolutional layer with 5d weight tensor that is not contiguous

### ISSUE THREAD

**Issue by zbh2047** [pytorch/pytorch#32088]
## 🚀 Feature
Given a batch of images, having a shape (b, c, h, w)
Given a weight tensor of shape (b, out_features, in_features, kernel_height, kernel_width) which is not contiguous (can not view as (b * out_features, in_features, kernel_height, kernel_width) without copying)
Perform a convolution that uses each 4d weight tensor to convolve the corresponding 3d image.

## Motivation
For example, if one want to calculate the Jacobian matrix (i.e. the gradient with respect to parameters for each input), he can first create a 5d weight tensor using torch.expand() efficiently, 
 then perform batch convolutional layer, and finally backpropagate. In such case, weight tensor is not contiguous.
For another applications, consider an encoder-decoder network. If an encoder layer shares the same convolutional kernel with the decoder layer, then the weight tensor of the encoder layer is the transpose of  the decoder layer, thus not contiguous.

## Alternatives
Of course one can use torch.conv2d with group parameter groups=b. However, reshaping the 5d tensor to 4d (b * out_features, in_features, kernel_height, kernel_width) is needed. If the tensor is not contiguous, it will involve a copy operation which is really time consuming.

## Additional context
For implementing batch convolutional layer using conv2d with group, see
https://github.com/pytorch/pytorch/issues/17983 .
